sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -137,28 +137,83 @@ object ResolveHints {
   }
 
   /**
-   * COALESCE Hint accepts name "COALESCE" and "REPARTITION".
-   * Its parameter includes a partition number.
+   * COALESCE Hint accepts name "COALESCE" and "REPARTITION" and "REPARTITIONBYRANGE".
+   * Its parameter includes a partition number and columns.
    */
   object ResolveCoalesceHints extends Rule[LogicalPlan] {
-    private val COALESCE_HINT_NAMES = Set("COALESCE", "REPARTITION")
+    private val COALESCE_HINT_NAMES = Set("COALESCE", "REPARTITION", "REPARTITIONBYRANGE")
+
+    /**
+     * same with repartition api
+     */
+    private def createRepartition(
+        shuffle: Boolean, h: UnresolvedHint): LogicalPlan = {
+      def createRepartitionByExpression(
+           numPartitions: Int, exprs: Seq[Any], h: UnresolvedHint): RepartitionByExpression = {
+        val errExprs = exprs.filter(!_.isInstanceOf[UnresolvedAttribute])
```

Comment:
We cannot match `Column` here because of module dependency.

Reply:
How about the case below?
```
scala> df.hint("repartition", 1, df("a")).groupBy("a").sum().explain(true)
org.apache.spark.sql.AnalysisException: Repartition hint parameter should be columns but was ArrayBuffer(a#10);
  at org.apache.spark.sql.catalyst.analysis.ResolveHints$ResolveCoalesceHints$.createRepartitionByExpression$1(ResolveHints.scala:154)
  at org.apache.spark.sql.catalyst.analysis.ResolveHints$ResolveCoalesceHints$.org$apache$spark$sql$catalyst$analysis$ResolveHints$ResolveCoalesceHints$$createRepartition(ResolveHints.scala:169)
```